packages/sdk/src/types/runtime/behaviors/action.ts

```diff
@@ -24,13 +25,34 @@ export class DiscreteAction {
 	private handlers: ActionHandlers = {};
 	private activeUserIds: string[] = [];
 
+	/** @hidden */
+	constructor(private name: string) {
+	}
+
 	/**
+	 * @hidden
 	 * Add a handler for the given action state for when it is triggered.
+	 * @param actorId The actor associated with this action.
 	 * @param actionState The action state that the handle should be assigned to.
 	 * @param handler The handler to call when the action state is triggered.
 	 */
-	public on(actionState: ActionState, handler: ActionHandler): this {
+	public on(
+		sendPayload: (payload: Payloads.Payload) => void,
+		actorId: string,

```

Comment:
I like this solution better than the whole passing in the context.  This does seem to be becoming more of a pattern where we want to send specific payloads from elsewhere besides in the actor and the context itself.  Seems we might want to consider a more elegant solution in the long run.  Thoughts?